### PR TITLE
Fixes #836  extract unit and and name

### LINKED ISSUE #836: export extractNameAndUnit method

### PR DESCRIPTION
**Pull request by msevestre**
Fixes #836

### CODE REVIEW

File: tests/testthat/test-utilities-units.R [hunk 1]
```diff
@@ -137,8 +137,39 @@ test_that("It returns null if the dimension is not found for the unit", {
   expect_null(getDimensionForUnit("toto"))
 })
 
-context("getUnitsForDimension")
 
+context("extractNameAndUnit")
+test_that("It can extract name and unit when no unit is present", {
+  res <- extractNameAndUnit("Value")
```
[the remark this answers is not in the document]
Comment (msevestre) [reply tests/testthat/test-utilities-units.R:143]: it returns your input string

File: R/utilities-units.R [hunk 2]
```diff
@@ -287,6 +288,21 @@ getDimensionByName <- function(name) {
   rClr::clrCall(dimensionTask, "DimensionByName", enc2utf8(name))
 }
 
+#' Returns a named list with two entries (name, unit) corresponding to the name and unit
+#' extracted out of the `text` provided as parameter
+#'
+#' @examples
+#' res <- extractNameAndUnit("Value [mg]")
+#' # res$name == "Value", res$unit == "mg"
```
Comment (msevestre) [inline R/utilities-units.R:296]: @PavelBal This example does not help?
How would you name this? The goal of the implementation is to be able to split a string and return the variable part (name) and the unit part (unit)
Say your text is "Concentration [mg]". It returns "Concentration" and "mg" a

**Comment by msevestre** [comment]
I ran the styler in this file as well. (News.md)

**Comment by PavelBal** [comment]
`extractUnitFromColumnHeader` or something like that as an alternative name?

To be honest, I do not hink this function belongs to the core package. It seems to be tailored for specific use cases (I guess ReportingEngine?) and I cannot image a general use case for it, especially as we provide methods for reading excel data files using the importer.

**Comment by PavelBal** [comment]
I understand now what it does, but I stilll think it does not belong here. Why cannot it be moved to the ReportingEngine? It does not require the DotNetWrapper, as you are directly calling rClr, and the ReportingEngine will always require rClr being loaded.

**Comment by msevestre** [comment]
I see. Yes we could do this indeed.

**Comment by pchelle** [comment]
> @pchelle Can you please look at this PR and copy the tests and the code in lieu of your current implementation?

@msevestre Do I also need to add the dll somewhere ? because I get the following error message when trying to implement the method:
```r
Error in rClr::clrCall(dimensionTask, "ExtractNameAndUnit", enc2utf8(text)) : 
  Type:    System.MissingMethodException
Message: Could not find a suitable instance method ExtractNameAndUnit on type OSPSuite.R.Services.DimensionTask for method parameters String
Method:  Void ThrowMissingMethod(System.Type, System.String, System.String, System.Type[])
Stack trace:
   at Rclr.ReflectionHelper.ThrowMissingMethod(Type classType, String methodName, String modifier, Type[] types)
   at Rclr.ClrFacade.InternalCallInstanceMethod(Object obj, String methodName, Boolean tryUseConverter, Object[] arguments)
```

**Comment by PavelBal** [comment]
@pchelle make sure you are using the latest ospsuite-r.

**Comment by msevestre** [comment]
Well the DLL is not in the lastest ospsuite-r. Need to update this at list

**Comment by msevestre** [comment]
@pchelle This is building

**Comment by PavelBal** [comment]
@msevestre Delete the branch?

**Pull request closed without merging**